hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoopImpl/mapreduce/lib/ConfiguratorBase.java

```diff
@@ -102,69 +78,26 @@ protected static String enumToConfKey(Enum<?> e) {
-      try (AccumuloClient client = Accumulo.newClient().from(info.getProperties()).build()) {
-        AuthenticationToken token = client.securityOperations()
-            .getDelegationToken(new DelegationTokenConfig());
-        result = ClientInfo.from(Accumulo.newClientProperties().from(info.getProperties())
-            .as(info.getPrincipal(), token).build());
-      } catch (Exception e) {
-        log.warn("Failed to automatically obtain DelegationToken, "
-            + "Mappers/Reducers will likely fail to communicate with Accumulo", e);
-      }
-    }
-    // DelegationTokens can be passed securely from user to task without serializing insecurely in
-    // the configuration
-    if (info.getAuthenticationToken() instanceof DelegationTokenImpl) {
-      DelegationTokenImpl delegationToken = (DelegationTokenImpl) info.getAuthenticationToken();
-
-      // Convert it into a Hadoop Token
-      AuthenticationTokenIdentifier identifier = delegationToken.getIdentifier();
-      Token<AuthenticationTokenIdentifier> hadoopToken = new Token<>(identifier.getBytes(),
-          delegationToken.getPassword(), identifier.getKind(), delegationToken.getServiceName());
-
-      // Add the Hadoop Token to the Job so it gets serialized and passed along.
-      credentials.addToken(hadoopToken.getService(), hadoopToken);
-    }
-    return result;
-  }
-
-  public static void setClientInfo(Class<?> implementingClass, Configuration conf,
-      ClientInfo info) {
-    setClientProperties(implementingClass, conf, info.getProperties());
-    conf.setBoolean(enumToConfKey(implementingClass, ConnectorInfo.IS_CONFIGURED), true);
-  }
-
-  public static ClientInfo getClientInfo(Class<?> implementingClass, Configuration conf) {
-    Properties props = getClientProperties(implementingClass, conf);
-    return new ClientInfoImpl(props);
-  }
-
-  public static void setClientPropertiesFile(Class<?> implementingClass, Configuration conf,
-      String clientPropertiesFile) {
-    try {
-      DistributedCacheHelper.addCacheFile(new URI(clientPropertiesFile), conf);
-    } catch (URISyntaxException e) {
-      throw new IllegalStateException("Unable to add client properties file \""
-          + clientPropertiesFile + "\" to distributed cache.");
-    }
-    conf.set(enumToConfKey(implementingClass, ClientOpts.CLIENT_PROPS_FILE), clientPropertiesFile);
-    conf.setBoolean(enumToConfKey(implementingClass, ConnectorInfo.IS_CONFIGURED), true);
-  }
-
   public static void setClientProperties(Class<?> implementingClass, Configuration conf,
-      Properties props) {
-    StringWriter writer = new StringWriter();
-    try {
-      props.store(writer, "client properties");
-    } catch (IOException e) {
-      throw new IllegalStateException(e);
+      Properties props, String clientPropsPath) {
+    if (clientPropsPath != null) {
+      try {
+        DistributedCacheHelper.addCacheFile(new URI(clientPropsPath), conf);
```

Comment:
Should update the DistributedCacheHelper to not use deprecated methods so it can be un-deprecated.  This could be a separate PR though

Reply:
I need something that works with both mapreduce & mapred.  I am not sure what a better option is.  We might be able to do the following but I think this should be a separate issue (I will create one):
```java
    String files = conf.get("mapreduce.job.cache.files");
    conf.set("mapreduce.job.cache.files", files == null ? uri.toString() : files + "," + uri.toString());
```